### ISSUE TITLE
API for add_corey_oil() and similar functions

### ISSUE THREAD

**Issue by berland** [equinor/pyscal#10]
How should the API for `add_corey_oil()` be? To simplify usage in interactive sessions, it is tempting to add `**kwargs` to allow for sending in a dict where only a subset of the args are used, then one can do:

```python
p = dict(swl=.., sorg=.., nw=.., now=.., ng=.., etc)
wog = WaterOilGas(p)
wog.wateroil.add_corey_oil(p)
wog.wateroil.add_corey_water(p)
wog.gasoil.add_corey_gas(p)
wog.gasoil.add_corey_oil(p)
```

If ``kwargs`` is not there to swallow superfluous arguments, we need to be explicit:
```python
wog.wateroil.add_corey_water(nw=p['nw'], krwend=p['krwend'], krwmax=p['krwmax'])
```
(and we cannot use the default values provided in`add_corey_water` function signature)

This is neat, but allowing deliberately superfluous arguments to a function might not be good practice.

**Comment by berland** [comment]
Generalized issue:
```
def func_a(x1=1, x2=2, **kwargs):
    """
    only x1 and x2 are used, remainder kwargs ignored
    """
    print((x1, x2))
    
# Another function with some argument overlap:
def func_b(x2=2, x3=3, **kwargs):
    """
    only x2 and x3 are used, remainder kwargs ignored
    """
    print((x2, x3))
    
# A dictionary with parameters to be used by the func_*s
args = dict(x1=10, x2=20, x3=30)

# Allows simple API, but you don't see exactly what is being used here:
func_a(**args)
func_b(**args)

# This does not work - ok!
# func_a(x2=200, **args)  # TypeError, multiple values

# Explicit way that does not require **kwargs in the def's:
def subdict(d, subkeys):
    """Return a subset of a dictionary"""
    return dict((k, d[k]) for k in subkeys)

# Make a version of func_a with stricter input
def func_a_strict(x1=1, x2=2):
    func_a(x1, x2)
    

func_a_strict(**subdict(args, ['x1', 'x2']))
# Equivalent:
func_a_strict(x1=args['x1'], x2=args['x2'])
```

**Comment by berland** [comment]
kwargs here breaks the Zen of Python: Explicit is better than implicit

**Comment by berland** [comment]
Another alternative could be to use the Factory pattern for constructing WaterOilGas objects.

**Comment by berland** [comment]
See #11

**Comment by berland** [comment]
Closing as the factory class has been merged in #11
